### PR TITLE
Document nint and nuint

### LINKED ISSUE #20582: Language reference: nint, nuint
The builtin types page should add `nint` and nuint`

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #20582

### CODE REVIEW

File: docs/csharp/language-reference/builtin-types/built-in-types.md [hunk 1]
```diff
@@ -22,11 +21,15 @@ The following table lists the C# built-in [value](value-types.md) types:
 |[`float`](floating-point-numeric-types.md)|<xref:System.Single?displayProperty=nameWithType>|
 |[`int`](integral-numeric-types.md)|<xref:System.Int32?displayProperty=nameWithType>|
 |[`uint`](integral-numeric-types.md)|<xref:System.UInt32?displayProperty=nameWithType>|
+|[`nint`](integral-numeric-types.md)|<xref:System.IntPtr?displayProperty=nameWithType> \*|
+|[`nuint`](integral-numeric-types.md)|<xref:System.UIntPtr?displayProperty=nameWithType> \*|
```
Comment (pkulikov) [inline docs/csharp/language-reference/builtin-types/built-in-types.md:25]: Why not to reference the article about native integers directly here?

Comment (tdykstra) [reply docs/csharp/language-reference/builtin-types/built-in-types.md]: Good idea, I'll do that.

Comment (pkulikov) [reply docs/csharp/language-reference/builtin-types/built-in-types.md]: Ah, sorry @tdykstra I was not clear. I've meant the links in the left-most column.

File: docs/csharp/language-reference/builtin-types/built-in-types.md [hunk 2]
```diff
@@ -22,11 +21,15 @@ The following table lists the C# built-in [value](value-types.md) types:
...
 |[`int`](integral-numeric-types.md)|<xref:System.Int32?displayProperty=nameWithType>|
 |[`uint`](integral-numeric-types.md)|<xref:System.UInt32?displayProperty=nameWithType>|
+|[`nint`](nint-nuint.md)|<xref:System.IntPtr?displayProperty=nameWithType>\*|
+|[`nuint`](nint-nuint.md)|<xref:System.UIntPtr?displayProperty=nameWithType>\*|
 |[`long`](integral-numeric-types.md)|<xref:System.Int64?displayProperty=nameWithType>|
 |[`ulong`](integral-numeric-types.md)|<xref:System.UInt64?displayProperty=nameWithType>|
 |[`short`](integral-numeric-types.md)|<xref:System.Int16?displayProperty=nameWithType>|
 |[`ushort`](integral-numeric-types.md)|<xref:System.UInt16?displayProperty=nameWithType>|
 
+\* For more information, see [`nint` and `nuint` types](nint-nuint.md).
```
Comment (pkulikov) [inline docs/csharp/language-reference/builtin-types/built-in-types.md:31]: Now, with the links updated in the table, this is not necessary.

File: docs/csharp/language-reference/builtin-types/nint-nuint.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+The native-sized integer types are represented internally as the .NET types <xref:System.IntPtr?displayProperty=nameWithType> and <xref:System.UIntPtr?displayProperty=nameWithType>. Unlike other numeric types, the keywords are not simply aliases for the types. The following statements are not equivalent:
+
+```csharp
+nint a = 1;
+System.IntPtr a = 1;
+```
+
+The compiler provides operations and conversions for `nint` and `nuint` that are appropriate for integer types.
+
+## Run-time integer size
```
Comment (BillWagner) [inline docs/csharp/language-reference/builtin-types/nint-nuint.md:25]: I'm torn, but wonder about specifying native integer size here:

```suggestion
## Run-time native integer size
```

File: docs/csharp/language-reference/builtin-types/nint-nuint.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+* For `nint`: <xref:System.Int32.MinValue?displayProperty=nameWithType> to <xref:System.Int32.MaxValue?displayProperty=nameWithType>.
+* For `nuint`: <xref:System.UInt32.MinValue?displayProperty=nameWithType> to <xref:System.UInt32.MaxValue?displayProperty=nameWithType>.
+
+## Conversions
+
+The compiler provides implicit and explicit conversions to other numeric types. For more information, see [Built-in numeric conversions](numeric-conversions.md).
+
+## Literals
+
+There's no direct syntax for native-sized integer literals. You can use implicit or explicit casts of other integer values instead. For example:
```
Comment (BillWagner) [inline docs/csharp/language-reference/builtin-types/nint-nuint.md:52]: I think this is needs clarification:

```suggestion
There's no direct syntax for native-sized integer literals. There's no suffix to indicate that a literal is a native-sized integer, such as `L` to indicate a `long`. You can use implicit or explicit casts of other integer values instead. For example:
```

**Review by BillWagner — APPROVED** [review]
This is great @tdykstra 

I had a couple questions. Once you look at those, you can :shipit:

File: docs/csharp/language-reference/builtin-types/nint-nuint.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+```
+
+## Unsupported IntPtr/UIntPtr members
+
+The following members of <xref:System.IntPtr> and <xref:System.UIntPtr> aren't supported for `nint` and `nuint` types:
+
+* Parameterized constructors
+* <xref:System.IntPtr.Add(System.IntPtr,System.Int32)>
+* <xref:System.IntPtr.CompareTo%2A>
+* <xref:System.IntPtr.Size> - Use [sizeOf()](#run-time-integer-size) instead. Although `nint.Size` isn't supported, you can use `IntPtr.Size` to get an equivalent value.
```
Comment (tdykstra) [inline docs/csharp/language-reference/builtin-types/nint-nuint.md:66]:
```suggestion
* <xref:System.IntPtr.Size> - Use [sizeOf()](#run-time-native-integer-size) instead. Although `nint.Size` isn't supported, you can use `IntPtr.Size` to get an equivalent value.
```

**Pull request merged**